### PR TITLE
Docs updated for buttons

### PR DESCRIPTION
**Pull request by raframil**
Improved documentation regarding buttons, their actions and how to use them in other ways with Angular

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -201,6 +201,70 @@ this.shepherdService.addSteps([
...
+};
+```
+
+In this simple example, we have three buttons: each one with different types and classes. 
+
+**⚠️ You can't set up a type and an action at the same time inside a button**. 
+
+There are some techniques to make certain buttons perform the actions you want on the component. In Angular, you can use Decorators to make some actions work by events. For example, you can listen for document clicks (mouse events) and check what the user is doing.
+
+First, you need to import HostListener from @angular/core.
```
Comment (rwwagner90) [inline README.md]: I don't think we should document this stuff. You can pass `action` to your buttons for custom actions, no need to do all this, right?

Comment (raframil) [reply README.md]: I agree with you, but if you use a separated js file from your component, you can't access different services. For custom actions inside angular components this is a workaround. Not sure if it should be in shepherd readme, let me know. I may write more about action

Comment (rwwagner90) [reply README.md]: I'm saying that the `action` needs to be a function. Where the action comes from can vary based on the framework and setup of your app.

File: README.md [hunk 2]
```diff
@@ -225,45 +226,23 @@ const builtInButtons = {
...
 ```
 
-In this simple example, we have three buttons: each one with different types and classes. 
-
-**⚠️ You can't set up a type and an action at the same time inside a button**. 
-
-There are some techniques to make certain buttons perform the actions you want on the component. In Angular, you can use Decorators to make some actions work by events. For example, you can listen for document clicks (mouse events) and check what the user is doing.
-
-First, you need to import HostListener from @angular/core. 
+Buttons can have an action inside then. An action expects a function, and you can set whatever you need inside this. Whenever the button is clicked, the function will be executed. You can use it for default shepherd functions, like `this.shepherdService.complete()` or `this.shepherdService.next()`.It's possible to create your own function and set it inside an action.
```
Comment (rwwagner90) [inline README.md]:
```suggestion
Buttons have an action property, which must be a function. Whenever the button is clicked, the function will be executed. You can use it for default shepherd functions, like `this.shepherdService.complete()` or `this.shepherdService.next()`, or create your own function to use for the action.
```

**Comment by rwwagner90** [comment]
Thanks so much for the help with the docs!

**Pull request merged**